front/src/components/header/subcomponents/LoginModal.tsx

```diff
@@ -92,56 +116,136 @@ const LoginModal = ({ disclosureProps, setIsLoggedIn, unregisteredClearFollowedS
-
           <ModalBody pb={6}>
             <FormControl isInvalid={Boolean(errors.email)}>
               <FormLabel htmlFor="email">Email</FormLabel>
-              <Input name="email" placeholder="Email" ref={register(formSchema.email)} autoFocus />
+              <Input
+                isDisabled={formOption === 0 || formOption === 1 ? false : true}
+                name="email"
+                placeholder="Email"
+                ref={register(formSchema.email)}
+                autoFocus
+              />
               <FormErrorMessage>{errors.email?.message}</FormErrorMessage>
             </FormControl>
 
-            <FormControl isInvalid={Boolean(errors.password)} mt={4}>
-              <FormLabel>Password</FormLabel>
-              <InputGroup>
-                <Input
-                  name="password"
-                  placeholder="Password"
-                  ref={register(formSchema.password)}
-                  type={passwordVisible ? 'text' : 'password'}
-                />
-                <InputRightElement width="4.5rem">
-                  <Button h="1.75rem" onClick={togglePasswordVisible} size="sm" tabIndex={-1}>
-                    {passwordVisible ? 'Hide' : 'Show'}
-                  </Button>
-                </InputRightElement>
-              </InputGroup>
-              <FormErrorMessage>{errors.password?.message}</FormErrorMessage>
-            </FormControl>
-
+            {(formOption === 0 || formOption === 3) && (
+              <FormControl isInvalid={Boolean(errors.password)} mt={4}>
+                <FormLabel> {formOption === 3 && 'New'} Password</FormLabel>
+                <InputGroup>
+                  <Input
+                    name="password"
+                    placeholder="Password"
+                    ref={register(formSchema.password)}
+                    type={passwordVisible ? 'text' : 'password'}
+                  />
+                  <InputRightElement width="4.5rem">
+                    <Button h="1.75rem" onClick={togglePasswordVisible} size="sm" tabIndex={-1}>
+                      {passwordVisible ? 'Hide' : 'Show'}
+                    </Button>
+                  </InputRightElement>
+                </InputGroup>
+                <FormErrorMessage>{errors.password?.message}</FormErrorMessage>
+              </FormControl>
+            )}
+            {formOption === 2 && (
+              <FormControl isInvalid={Boolean(errors.password)} mt={4}>
+                <FormLabel>Enter Verification Code</FormLabel>
+                <Input name="otp" placeholder="OTP" ref={register(formSchema.otp)} />
+                <FormErrorMessage>{errors.otp?.message}</FormErrorMessage>
+              </FormControl>
+            )}
             <FormControl isInvalid={Boolean(errors.login)} mt={4}>
               <FormErrorMessage>{errors.login?.message}</FormErrorMessage>
             </FormControl>
+            <Link color="teal.500" onClick={() => setformOption((formOption + 1) % 2)}>
```

Comment:
I think the `(formOption + 1) % 2)` could be confusing - pls add a comment about what it's doing